mmv1/third_party/terraform/tests/resource_compute_instance_template_test.go.erb

```diff
@@ -2305,3 +2334,60 @@ resource "google_compute_resource_policy" "foo" {
 }
 `, suffix, policyName)
 }
+
+func testAccComputeInstanceTemplate_nictype(image, instance, nictype string) string {
+	return fmt.Sprintf(`
+resource "google_compute_image" "example" {
+	name = "%s"
+	raw_disk {
+		source = "https://storage.googleapis.com/bosh-gce-raw-stemcells/bosh-stemcell-97.98-google-kvm-ubuntu-xenial-go_agent-raw-1557960142.tar.gz"
+	}
+
+	guest_os_features {
+		type = "SECURE_BOOT"
+	}
+
+	guest_os_features {
+		type = "MULTI_IP_SUBNET"
+	}
+
+	guest_os_features {
+		type = "GVNIC"
```

Comment:
Should this always have the same type as the `network_interface.nic_type` or is it just a coincidence that they're the same?

Reply:
@melinath  Not always. When the **nic_type** is set to _GVNIC_  for compute instance or instance template, it does require _GVNIC_ is available for **guest_os_features** on the image. If the **nic_type** is set to another type, there is no such requirement.